kyaml/order/syncorder_test.go

```diff
@@ -0,0 +1,397 @@
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+func TestSyncOrder(t *testing.T) {
+	testCases := []struct {
+		name     string
+		from     string
+		to       string
+		expected string
+	}{
+		{
+			name: "sort data fields configmap with comments",
+			from: `apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: setters-config
+data:
+  # This should be the name of your Config Controller instance
+  cluster-name: cluster-name
+  # This should be the project where you deployed Config Controller
+  project-id: project-id # pro
+  project-number: "1234567890123"
+  # You can leave these defaults
+  namespace: config-control
+  deployment-repo: deployment-repo
+  source-repo: source-repo
+`,
+			to: `apiVersion: v1
+kind: ConfigMap
+metadata: # kpt-merge: /setters-config
+  name: setters-config
+data:
+  # You can leave these defaults
+  namespace: config-control
+  # This should be the name of your Config Controller instance
+  cluster-name: cluster-name
+  deployment-repo: deployment-repo
+  # This should be the project where you deployed Config Controller
+  project-id: project-id # project
+  project-number: "1234567890123"
+  source-repo: source-repo
+`,
+			expected: `apiVersion: v1
+kind: ConfigMap
+metadata: # kpt-merge: /setters-config
+  name: setters-config
+data:
+  # This should be the name of your Config Controller instance
+  cluster-name: cluster-name
+  # This should be the project where you deployed Config Controller
+  project-id: project-id # project
+  project-number: "1234567890123"
+  # You can leave these defaults
+  namespace: config-control
+  deployment-repo: deployment-repo
+  source-repo: source-repo
+`,
+		},
+		{
+			name: "sort data fields configmap but retain order of extra fields",
+			from: `apiVersion: v1
+kind: ConfigMap
+data:
+  baz: bar
+  cluster-name: cluster-name
+  foo: config-control
+`,
+			to: `kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: foo
+data:
+  color: orange
+  foo: config-control
+  abc: def
+  cluster-name: cluster-name
+`,
+			expected: `apiVersion: v1
+kind: ConfigMap
+data:
+  cluster-name: cluster-name
+  foo: config-control
+  color: orange
+  abc: def
+metadata:
+  name: foo
+`,
+		},
+		{
+			name: "sort containers list node with sequence head comments preservation",
+			from: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: before
+spec:
+  containers:
+  - name: nginx
+    # nginx image
+    image: "nginx:1.16.1"
+    ports:
+    - protocol: TCP # tcp protocol
+      containerPort: 80
+`,
+			to: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: after
+spec:
+  containers:
+  # ports comment
+  - ports:
+    - containerPort: 80
+      protocol: TCP # tcp protocol
+    # nginx image
+    image: "nginx:1.16.2"
+    # nginx container
+    name: nginx
+# end of resource
+`,
+			expected: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: after
+spec:
+  containers:
+  # ports comment
```

Comment:
I would have expected this comment should come right before `ports:` since that's where it is in the input. Do you think that makes sense, and is it possible to do?